plugins/woocommerce/tests/legacy/unit-tests/rest-api/Helpers/OrderHelper.php

```diff
@@ -126,4 +128,95 @@ public static function create_order( $customer_id = 1, $product = null ) {
 
 		return $order;
 	}
+
+	/**
+	 * Helper method to create custom tables if not present.
+	 */
+	public static function create_order_custom_table_if_not_exist() {
+		$order_table_controller = wc_get_container()
+			->get( 'Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController' );
+		$order_table_controller->show_feature();
+		$synchronizer = wc_get_container()
+			->get( DataSynchronizer::class );
+		if ( ! $synchronizer->check_orders_table_exists() ) {
+			$synchronizer->create_database_tables();
+		}
```

Comment:
I wonder if this is slightly redundant ... 

- `$synchronizer->check_orders_table_exists()` (possibly a slightly misleading method name) does not just check that the tables exist, it tries to create the tables if they are missing:
    -  That's because it calls `::get_missing_tables()` (also might be somewhat misleading, because it doesn't hint at the side effects) ...
    - ... Which in turn calls `::dbDelta()`
- If tables are still missing at the end of that we are trying again anyway with the further call to `::create_database_tables()`, which itself just calls `::dbDelta()`.

I guess that's fine if we're consciously trying a second time but, because of the method names, that's not quite how it reads (it seems like we're unintentionally trying twice). Otherwise, I wonder if:

- We should rename `::check_orders_table_exists()` to something like `::ensure_order_tables_exist()`, and;
- If it fails, log an error (or maybe an error should be logged by that method itself).

Reply:
This is fine actually because `dbDelta` is called with the `execute` param set to false. In this case, it will try to generate all the queries it would have run but does not execute them. So it's more like a dry name, thereby making the name of these methods accurate.